### PR TITLE
esther/GP-97/linebot-voice-to-text

### PR DESCRIPTION
**Pull request by Esther125**
## Test
- follow the steps in the docs to **rebuild** and run docker and set up chatbot (start ngrok, update webhook url...)
- send an audio message to the group that includes our official account

## Note
- there's a conflict between black and isort, and therefore I turned off the isort pre-commit in main.py
- taiwanese voice recognition is not done yet. We can open a new ticket for this function.

### CODE REVIEW

File: src/chatbot/services.py [hunk 1]
```diff
@@ -1,8 +1,14 @@
+# flake8:noqa
 import os
 from datetime import datetime
+from io import BytesIO
 
 import requests
+import speech_recognition as sr
```
Comment (yun-si) [inline src/chatbot/services.py:7]: Maybe the import path need to be complete, such as from src/...

**Review by yun-si — CHANGES_REQUESTED** [review]
when I run `uvicorn src.chatbot.main:app --host 0.0.0.0 --port 8080 --reload`
I met the error below
```
INFO:     Will watch for changes in these directories: ['/srv/graduation-project']
INFO:     Uvicorn running on http://0.0.0.0:8080 (Press CTRL+C to quit)
INFO:     Started reloader process [72] using WatchFiles
Process SpawnProcess-1:
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/multiprocessing/process.py", line 314, in _bootstrap
    self.run()
  File "/usr/local/lib/python3.10/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/local/lib/python3.10/site-packages/uvicorn/_subprocess.py", line 80, in subprocess_started
    target(sockets=sockets)
  File "/usr/local/lib/python3.10/site-packages/uvicorn/server.py", line 65, in run
    return asyncio.run(self.serve(sockets=sockets))
  File "/usr/local/lib/python3.10/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "uvloop/loop.pyx", line 1517, in uvloop.loop.Loop.run_until_complete
  File "/usr/local/lib/python3.10/site-packages/uvicorn/server.py", line 69, in serve
    await self._serve(sockets)
  File "/usr/local/lib/python3.10/site-packages/uvicorn/server.py", line 76, in _serve
    config.load()
  File "/usr/local/lib/python3.10/site-packages/uvicorn/config.py", line 434, in load
    self.loaded_app = import_from_string(self.app)
  File "/usr/local/lib/python3.10/site-packages/uvicorn/importer.py", line 22, in import_from_string
    raise exc from None
  File "/usr/local/lib/python3.10/site-packages/uvicorn/importer.py", line 19, in import_from_string
    module = importlib.import_module(module_str)
  File "/usr/local/lib/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/srv/graduation-project/src/chatbot/main.py", line 13, in <module>
    from src.chatbot.services import (
  File "/srv/graduation-project/src/chatbot/services.py", line 7, in <module>
    import speech_recognition as sr
ModuleNotFoundError: No module named 'speech_recognition'
```

File: src/chatbot/services.py (hunk 1 above)
Comment (Esther125) [reply src/chatbot/services.py:7]: speech_recognition is a package not a file
maybe you need to rebuild the docker image

Comment (yun-si) [reply src/chatbot/services.py:7]: sorry😅
maybe next time you can emphasize that we need to run `docker-compose build`~

Comment (Esther125) [reply src/chatbot/services.py:7]: sure!

**Comment by frafishen** [comment]
update this [line](https://github.com/Fang-4-Group/Graduation-Project/blob/ba232e82a0ac71006bcd28a00b2a21c2d09ef957/docs/08-Chatbot-Setup.md#L23) in the readme to 
```bash
MONGO_DB_URI=mongodb://mongodb:27017
```

**Comment by frafishen** [comment]
LINE_capture_743196440 987151
i also faced this QQ

**Review by yun-si — APPROVED** [review]
success~

**Pull request merged**